lib/interval/interval.ex

```diff
@@ -382,6 +382,71 @@ defmodule Timex.Interval do
   def max(%__MODULE__{until: until, right_open: false}), do: until
   def max(%__MODULE__{until: until}), do: Timex.shift(until, microseconds: -1)
 
+  @doc """
+  Returns an Interval representing the intersection between two intervals.
+  If the intervals do not overlap, return {:error, :no_overlap_interval}.
+  If the intervals overlap at a single instant (regardless of open/closed
+  bounds), also return {:error, :no_overlap_interval}
```

Comment:
Perhaps it should return `0` in the case of no overlap rather than an error? Two intervals not overlapping doesn't seem like an error to me.